calculator.rb

```diff
@@ -0,0 +1,67 @@
+# create a hash to store the operation and welcoming user
+puts "Welcome to Sharon's calculator Program! Please see below a list of operators you can choose from"
+accepted_operation = { "add" => "+", "subtract" => "-" , "multiply" => "*", "divide" => "/", "exponents" => "^" , "modulo" => "%" }
+
+# loop thru the operator hash in order to show user what the choices are
+i = 1
+accepted_operation.each do |key, value|
+    puts "#{i}. #{value} (#{key})" 
+    i += 1
+end
+
+# ask user for a math operation
+puts "Please chose one operator (either name or symbol)"
+operation = gets.chomp.downcase
+
+# test the operation that user input is valid or not
+until accepted_operation.values.include?(operation) || accepted_operation.keys.include?(operation.downcase)
+    puts "Invalid input! Please try again!"
+    operation = gets.chomp.downcase
+end
+
+# create method for checking if the user has input a valid integer number for the formular
+def num_check(user_input)
+   while (user_input != '0') && (user_input.to_i.to_s != user_input.strip && user_input.to_f.to_s != user_input.strip) do
+        puts "Invalid input, please enter an valid number!"
+        user_input = gets.chomp
+    end
+    return user_input.to_f
+end
+
+# ask user for the first number and check if it's valid
```

Comment:
You've written almost exactly the same code here twice, to get the first number and the second number. Could you DRY that up by putting this logic in a method or loop?